docs/src/main/asciidoc/websockets-next-reference.adoc

```diff
@@ -998,6 +998,66 @@ public class ProductEndpoint {
 <1> The `getProduct` callback method can only be invoked if the current security identity has an `admin` role or the user is allowed to get the product detail.
 <2> The error handler is invoked in case of the authorization failure.
 
+==== Bearer token authentication
+
+The xref:security-oidc-bearer-token-authentication.adoc[OIDC Bearer token authentication] expects that bearer tokens are passed in the `Authorization` header during the initial HTTP handshake.
+While we recommend to use WebSocket clients that support setting custom headers in the initial HTTP handshake, there are users that rely on https://websockets.spec.whatwg.org/#the-websocket-interface[WebSockets API].
+In this API, the HTTP `Sec-WebSocket-Protocol` request header is the only header that users can configure.
+We have seen users to take advantage of this header for purpose which it is not intended like in the example below:
+
+[source,javascript]
+----
+const token = getBearerToken()
+const socket = new WebSocket("wss://" + location.host + "/chat/" + username, ["bearer", token]); <1>
+----
+<1> Indicate 2 sub-protocols supported by the client, the `bearer` sub-protocol and the token sub-protocol.
+
+The WebSocket server can also be configured with supported sub-protocols:
+
+[source, properties]
+----
+quarkus.websockets-next.server.supported-subprotocols=bearer
+----
+<1> The `bearer` sub-protocol matches the sub-protocol option passed to the client in previous example.
+
+In Quarkus, it is possible to map HTTP server request headers to a different headers, for example:
+
+[source, java]
+----
+package io.quarkus.websockets.next.test.security;
+
+import static io.quarkus.websockets.next.HandshakeRequest.SEC_WEBSOCKET_PROTOCOL;
+import static io.vertx.core.http.HttpHeaders.AUTHORIZATION;
+
+import jakarta.enterprise.event.Observes;
+
+import io.vertx.ext.web.Router;
+
+public class WebSocketTokenHeaderHandler {
+
+    private static final String BEARER_PROTOCOL = "bearer";
+    private static final String TOKEN_PREFIX = BEARER_PROTOCOL + ",";
+
+    void observe(@Observes Router router) {
```

Comment:
It should be also possible to manipulate the request headers in `HttpUpgradeCheck.perform(HttpUpgradeContext)`, or is it too late for security?

Reply:
yes, it's too late

Reply:
I see.

Reply:
Do you have in mind something different than this PR, like only add section: here is how to use `HttpUpgradeCheck` to re-map headers and it works if you only use security annotations and disable proactive authentication?

Reply:
For people who has only WebSockets Next, I suspect it wouldn't have negative effect as annotations are likely their first option and proactive auth doesn't seem like important concept here.